cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -434,16 +562,21 @@ func (ht *hcsTask) Wait() *task.StateResponse {
 	return ht.init.Wait()
 }
 
-func (ht *hcsTask) waitInitExit() {
+func (ht *hcsTask) waitInitExit(destroyContainer bool) {
 	ctx, span := trace.StartSpan(context.Background(), "hcsTask::waitInitExit")
 	defer span.End()
 	span.AddAttributes(trace.StringAttribute("tid", ht.id))
 
 	// Wait for it to exit on its own
 	ht.init.Wait()
 
-	// Close the host and event the exit
-	ht.close(ctx)
+	if destroyContainer {
+		// Close the host and event the exit
+		ht.close(ctx)
+	} else {
+		// Only close the container without destroying it
```

Comment:
This is a bit confusing. The comment on `closeHost` says that it just closes the UVM and releases it's resources, if the task owns the host. I thinks something like 

```
// Close the container's host, but do not close or terminate the container itself. 
```